packages/core/src/codewhisperer/views/activeStateController.ts

```diff
@@ -32,17 +32,25 @@ export class ActiveStateController implements vscode.Disposable {
 
     constructor(private readonly container: Container) {
         this._disposable = vscode.Disposable.from(
-            RecommendationService.instance.suggestionActionEvent(this.onSuggestionActionEvent, this),
-            RecommendationHandler.instance.onDidReceiveRecommendation(this.onDidReceiveRecommendation, this),
-            this.container.lineTracker.onDidChangeActiveLines(this.onActiveLinesChanged, this),
-            subscribeOnce(this.container.lineTracker.onReady)(this.onReady, this),
+            RecommendationService.instance.suggestionActionEvent(async e => {
+                await this.onSuggestionActionEvent(e)
+            }),
+            RecommendationHandler.instance.onDidReceiveRecommendation(async _ => {
+                await this.onDidReceiveRecommendation()
+            }),
+            this.container.lineTracker.onDidChangeActiveLines(async e => {
+                await this.onActiveLinesChanged(e)
+            }),
+            subscribeOnce(this.container.lineTracker.onReady)(async _ => {
+                await this.onReady()
```

Comment:
FYI: If the eventemitter doesn't `await` the subscribed listeners, this will still silently fail. 

That's why sometimes it's better to `.catch()` and log the error instead of sending a `async` function to these eventemitters.

Reply:
understood, yea will play around with the approach you suggests Justin